lib/usd/ui/layerEditor/sessionState.h

```diff
@@ -41,21 +41,70 @@ class SessionState : public QObject
 {
     Q_OBJECT
 public:
-    virtual ~SessionState() { }
+    virtual ~SessionState() {}
 
     struct StageEntry
     {
         PXR_NS::UsdStageRefPtr _stage;
         std::string            _displayName;
         std::string            _proxyShapePath;
+
+        StageEntry()
+        {
+            _stage = PXR_NS::UsdStageRefPtr();
+            _displayName = "";
+            _proxyShapePath = "";
+        }
+
+        StageEntry(
+            PXR_NS::UsdStageRefPtr const& stage,
+            std::string const&            displayName,
+            std::string const&            proxyShapePath)
+        {
+            _stage = stage;
+            _displayName = displayName;
+            _proxyShapePath = proxyShapePath;
+        }
+
+        StageEntry(const StageEntry& entry)
+        {
+            _stage = entry._stage;
```

Comment:
did you need to declare all of these? We get all the copy contructors and equality operators for free when we don't define any.  It's a lot of code to maintain if we don't need them

Reply:
Yup normally I agree with you, but I needed the equality ones since Qt wasn't happy with the default. 

The only extra one's are the copy and other constructor, that was just habit and convenience, but I can remove them.